airflow/executors/base_executor.py

```diff
@@ -97,6 +100,15 @@ def queue_task_instance(
         cfg_path: Optional[str] = None,
     ) -> None:
         """Queues task instance."""
+        if mark_success is not None:
+            warnings.warn(
+                'The argument `mark_success` has been deprecated. Use `state=State.SUCCESS` instead.',
+                DeprecationWarning,
+            )
+            if mark_as is not None:
+                raise AirflowException('You cannot use both `mark_as` and `mark_success` arguments')
```

Comment:
```suggestion
                raise TypeError('You cannot use both `mark_as` and `mark_success` arguments')
```

Incorrect function signature should be type error.